test/test.config.js

```diff
@@ -7,29 +7,39 @@ import path from 'node:path';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
+const convertToBoolean = (envVariable = '') => {
+  if(/^(1|true)$/.test(envVariable)) {
+    return true;
+  }
+  return false;
+};
+
+const {connectOptions} = config.mongodb;
+
 // MongoDB
 config.mongodb.name = 'bedrock_mongodb_test';
 config.mongodb.host = process.env.MONGODB_HOST || 'localhost';
 config.mongodb.port = process.env.MONGODB_PORT || 27017;
-
-if(process.env.MONGODB_USERNAME && process.env.MONGODB_PASSWORD) {
-  config.mongodb.username = process.env.MONGODB_USERNAME;
-  config.mongodb.password = process.env.MONGODB_PASSWORD;
-  const {connectOptions} = config.mongodb;
-  // process.env.MONGODB_SSL should be 1 to test with SSL
-  connectOptions.ssl = Boolean(process.env.MONGODB_SSL) || false;
-  connectOptions.authSource = process.env.MONGODB_AUTHSOURCE || 'admin';
-  // this can safely be undefined
-  connectOptions.replicaSet = process.env.MONGODB_REPLICASET;
-  console.log(
-    'TESTING WITH AUTH ', {
-      username: config.mongodb.username,
-      authSource: connectOptions.authSource,
-      password: Boolean(config.mongodb.password),
-      ssl: connectOptions.ssl,
-      replicaSet: connectOptions.replicaSet
-    });
+// set the env variable to 1 or true to make these true
+// set the env variable to anything else to make them false
+if(process.env.MONGODB_SSL) {
+  connectOptions.ssl = convertToBoolean(process.env.MONGODB_SSL);
+}
+if(process.env.MONGODB_TLS) {
+  connectOptions.tls = convertToBoolean(process.env.MONGODB_TLS);
 }
+if(process.env.MONGODB_AUTHSOURCE) {
+  connectOptions.authSource = process.env.MONGODB_AUTHSOURCE;
+}
```

Comment:
Does all this "just work" for existing devs running stuff locally?

Reply:
This is test data that is only set for the tests and the local tests running without auth are fine. I did have to explicitly add `MONGODB_AUTHSOURCE=admin` so the github auth cli tests continue to work. That is more in line though with our production environment and also allows us to simulate having the wrong `authSource` quickly when debugging connection strings / options. Previously the `authSource` when testing defaulted to `admin` if `password` and `username` were provided: https://github.com/digitalbazaar/bedrock-mongodb/blob/issue-84-get-server-info-login/test/test.config.js

Reply:
Addressed here: https://github.com/digitalbazaar/bedrock-mongodb/pull/89/commits/3f4adaf8125c1a91910df261065578b738654944